backend/kesaseteli/applications/tests/test_youth_applications_api.py

```diff
@@ -112,6 +124,16 @@ def test_youth_application_post_valid_social_security_number(
     assert "social_security_number" in response.data
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize("running_number", list(range(10)))
+def test_youth_application_post_valid_random_data(api_client, running_number):
```

Comment:
I just wanted to run the function 10 times, I am open to suggestions for a better way to do this. So the running_number is just telling the running number of the function call.

Reply:
Ah okay, now I understand what that means. But why do you need to run the same test 10 times? I don't see the benefit of doing that and it just slows down the test running time. Unless your goal is to stress test the endpoint but I guess this is not the case

Reply:
I wanted to use a not-too-big number of random data values for testing this and chose an arbitrary number 10. It does slow down the test running time but not hugely, that's why I chose a smallish number. Does that help any to clarify my intent/idea?

Reply:
Oh if your intention is to test the endpoint with different random data generated by the factory then this method doesn't work 🙂  This is because every time the test changes its param, the random seed will be reset (because of [this](https://github.com/City-of-Helsinki/yjdh/pull/545/files/64389afe0e2073ee0f6dfc57bf30ea4d39065ddc#diff-fc241226e1bc2fabfd9dcd8ec4780d373484b3ff0acb20909b5dd3f841656aeeR11)) the factory will always generate the same data. You can verify my concern by printing out `youth_application` each time the test run. 
So in order to achieve your goal, a simple solution is just to use for-loop inside the test 10 times. Then you can have 10 different random `youth_application` data

Reply:
Hmm... actually I think I already tried it and it does not reset the seed for each parametrization of the same test. So it does generate AFAIK 10 different random YouthApplication objects and the same random 10 ones because of the:
```python
@pytest.fixture(autouse=True)
def setup_test_environment(settings):
    factory.random.reseed_random("888")
```
BUT **I can doublecheck later** if needed also, now I'm busy on another task and can't check.

Reply:
I doublechecked and yes, actually what you said is true, it does use the same seed on each parametrization. Well, you live and learn! I changed it to use different seed on each run so now it does what I wanted it to do:

```diff
diff --git a/backend/kesaseteli/applications/tests/test_youth_applications_api.py b/backend/kesaseteli/applications/tests/test_youth_applications_api.py
index 69a3b73c..f169eae8 100644
--- a/backend/kesaseteli/applications/tests/test_youth_applications_api.py
+++ b/backend/kesaseteli/applications/tests/test_youth_applications_api.py
@@ -1,3 +1,4 @@
+import factory.random
 import pytest
 from rest_framework import status
 from rest_framework.reverse import reverse
@@ -125,8 +126,9 @@ def test_youth_application_post_valid_social_security_number(


 @pytest.mark.django_db
-@pytest.mark.parametrize("running_number", list(range(10)))
-def test_youth_application_post_valid_random_data(api_client, running_number):
+@pytest.mark.parametrize("random_seed", list(range(10)))
+def test_youth_application_post_valid_random_data(api_client, random_seed):
+    factory.random.reseed_random(random_seed)
     youth_application = YouthApplicationFactory()
     data = YouthApplicationSerializer(youth_application).data
     response = api_client.post(reverse("v1:youthapplication-list"), data)
diff --git a/backend/kesaseteli/common/tests/factories.py b/backend/kesaseteli/common/tests/factories.py
index bfa99e52..df25a74f 100644
--- a/backend/kesaseteli/common/tests/factories.py
+++ b/backend/kesaseteli/common/tests/factories.py
@@ -144,7 +144,7 @@ class BaseYouthApplicationFactory(factory.django.DjangoModelFactory):
     school = factory.Faker("random_element", elements=get_all_test_schools())
     is_unlisted_school = factory.LazyAttribute(uses_unlisted_test_school)
     email = factory.Faker("email")
-    phone_number = factory.Faker("phone_number", locale="fi")
+    phone_number = factory.LazyFunction(get_test_phone_number)
     language = factory.Faker("random_element", elements=["fi", "sv", "en"])
     _is_active = None
```

Also fixed the generation of test phone numbers in the YouthApplicationFactory. This came up because now the random seeds were different and problematic test phone numbers were generated.

☝️ **Force pushed the above changes**